src/unittest/test_utilities.cpp

```diff
@@ -636,3 +638,69 @@ void TestUtilities::testSanitizeDirName()
 	UASSERT(sanitizeDirName("cOnIn$", "~") == "~cOnIn$");
 	UASSERT(sanitizeDirName(" cOnIn$ ", "~") == "_cOnIn$_");
 }
+
+template <typename F, typename C>
+C container_map(const C &co, F functor)
+{
+	C ret;
+	for (auto it = co.begin(); it != co.end(); it++)
+		ret.push_back(functor(*it));
+	return ret;
+}
+
+void TestUtilities::testIsBlockInSight()
+{
+	std::vector<v3s16> testdata1 = {
+		{0, 1 * (int)BS, 0}, // camera_pos
+		{1, 0, 0},           // camera_dir
+
+		{ 2, 0, 0},
+		{-2, 0, 0},
+		{0, 0,  3},
+		{0, 0, -3},
+		{0, 0, 0},
+		{6, 0, 0}
+	};
+	auto test1 = [] (const std::vector<v3s16> &data) {
+		float range = BS * MAP_BLOCKSIZE * 4;
+		float fov = 72 * core::DEGTORAD;
+		v3f cp = v3f(data[0].X, data[0].Y, data[0].Z),
+			cd = v3f(data[1].X, data[1].Y, data[1].Z);
+		UASSERT( isBlockInSight(data[2], cp, cd, fov, range));
+		UASSERT(!isBlockInSight(data[3], cp, cd, fov, range));
+		UASSERT(!isBlockInSight(data[4], cp, cd, fov, range));
+		UASSERT(!isBlockInSight(data[5], cp, cd, fov, range));
+
+		// camera block must be visible
+		UASSERT(isBlockInSight(data[6], cp, cd, fov, range));
+
+		// out of range is never visible
+		UASSERT(!isBlockInSight(data[7], cp, cd, fov, range));
+	};
+	// XZ rotations
+	for (int j = 0; j < 4; j++) {
+		auto tmpdata = container_map(testdata1, [&] (v3s16 v) -> v3s16 {
+			v.rotateXZBy(j*90);
+			return v;
+		});
+		test1(tmpdata);
+	}
+	// just two for XY
+	for (int j = 0; j < 2; j++) {
+		auto tmpdata = container_map(testdata1, [&] (v3s16 v) -> v3s16 {
+			v.rotateXYBy(90+j*180);
+			return v;
+		});
+		test1(tmpdata);
+	}
+
+
+	{
+		float range = BS * MAP_BLOCKSIZE * 4;
+		float fov = 72 * core::DEGTORAD;
+		v3f cp(-(MAP_BLOCKSIZE - 1) * BS, 0, 0), cd(1, 0, 0);
```

Comment:
```suggestion
		v3f cam_pos(-(MAP_BLOCKSIZE - 1) * BS, 0, 0),
			cam_dir(1, 0, 0);
```
Same for above to clarify the meaning of these variables.